node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -89,6 +98,51 @@ func NewOsDeviceConnectivityHelperScsiGeneric(executer executer.ExecuterInterfac
 	}
 }
 
+func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathdOutputByVolumeIds(volumeIdVariations []string) (string, error) {
+	return r.Helper.GetMpathdOutputByVolumeId(volumeIdVariations)
+}
+
+func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDeviceName(volumePath string) (string, error) {
+	procMountsFileContent, err := ioutil.ReadFile(procMountsFilePath)
+	if err != nil {
+		return "", err
+	}
+	procMounts := string(procMountsFileContent)
+
+	return r.Helper.GetMpathDeviceNameFromProcMounts(procMounts, volumePath)
+}
+
+func (r OsDeviceConnectivityHelperScsiGeneric) GetMatchingVolumeIdToMpathName(mpathdOutput string,
+	mpathDeviceName string, volumeIdVariations []string) (string, error) {
+
+	volumeId, _ := r.Helper.GetVolumeIdByMpathdName(mpathDeviceName, mpathdOutput)
+
+	dmPath := filepath.Join(devMapperPath, filepath.Base(strings.TrimSpace(mpathDeviceName)))
+	if volumeId != "" {
+		if r.Helper.IsDmPathMatchesVolumeId(dmPath, volumeIdVariations) {
```

Comment:
why can't we just compare `volumeId` to `volumeIdVariations` here?

Reply:
I wanted that it will be like [`GetMpathDevice`](https://github.com/IBM/ibm-block-csi-driver/blob/404803e569d4ac8cf3f0912719eec9ae49883d0f/node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go#L205) function

Reply:
1. this looks like code dup
2. the way I see it:
`GetMpathDevice` gets an id and needs to return its dm path, so it verifies that the path indeed leads to that id using sg inq.
`GetMatchingVolumeIdToMpathName` (which IMO should be called `GetVolumeIdOfMpathName`) should get an mpath name and return its volume id (it can verify that the mpath name indeed leads to that id using sg inq), but it doesn't need to know/compare to the ("expected") id that kubelet requested

Reply:
maybe even `GetVolumeIdOfMpathName` -> `GetMpathVolumeId`?
sorry for the ping pong

Reply:
thanks.
and similarly, what do you think about `GetVolumeIdOfMpathDeviceName` -> `ExtractVolumeId`?

Reply:
done